operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,174 @@
+type KueueIntegrations string
+
+const (
+	BatchJob        KueueIntegrations = "BatchJob"
+	RayJob          KueueIntegrations = "RayJob"
+	RayCluster      KueueIntegrations = "RayCluster"
+	JobSet          KueueIntegrations = "JobSet"
+	MPIJob          KueueIntegrations = "MPIJob"
+	PaddeJob        KueueIntegrations = "PaddeJob"
+	PyTorchJob      KueueIntegrations = "PyTorchJob"
+	TFJob           KueueIntegrations = "TFJob"
+	XGBoostJob      KueueIntegrations = "XGBoostJob"
+	AppWrappers     KueueIntegrations = "AppWrappers"
+	Pod             KueueIntegrations = "Pod"
+	Deployment      KueueIntegrations = "Deployment"
+	StatefulSet     KueueIntegrations = "StatefulSet"
+	LeaderWorkerSet KueueIntegrations = "LeaderWorkerSet"
+)
+
+// This is the GVK for an external framework.
+// Controller runtime requires this in this format
+// for api discoverability.
+type ExternalFramework struct {
+	// group of externalFramework
+	// must be a valid qualified name consisting of a lower-case alphanumeric string,
+	// and hyphens of at most 63 characters in length. The name must start and end with an alphanumeric character.
+	// The name may be optionally prefixed with a subdomain consisting of lower-case alphanumeric characters,
+	// hyphens and periods, of at most 253 characters in length.
+	// Each period separated segment within the subdomain must start and end with an alphanumeric character.
+	// The optional prefix and the name are separate by a forward slash (/).
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.qualifiedName().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Group string `json:"group"`
+	// resource of external framework
+	// must be a valid qualified name consisting of a lower-case alphanumeric string,
+	// and hyphens of at most 63 characters in length. The name must start and end with an alphanumeric character.
+	// The name may be optionally prefixed with a subdomain consisting of lower-case alphanumeric characters,
+	// hyphens and periods, of at most 253 characters in length.
+	// Each period separated segment within the subdomain must start and end with an alphanumeric character.
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.qualifiedName().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Resource string `json:"resource"`
+	// version is the version of the api
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Version string `json:"version"`
```

Comment:
I do know that the version string should be a valid DNS label as outlined in the upstream Kube API conventions: https://github.com/kubernetes/community/blob/35444da79dff9a448e7ecf24b277e5f71373840a/contributors/devel/sig-architecture/api-conventions.md?plain=1#L116-L118

Reply:
Lets add that validation then